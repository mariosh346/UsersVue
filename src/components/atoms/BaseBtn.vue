<template>
  <q-btn
    :icon="icon"
    :loading="loading"
    :label="label"
    :type="type"
    :color="color"
    v-bind="$attrs"
    @click="$emit('click', $event)"
  />
</template>

<script setup lang="ts">
import type { QBtnProps } from 'quasar';

defineProps<QBtnProps>();

defineEmits<{
  (e: 'click', event: Event): void;
}>();
</script>
