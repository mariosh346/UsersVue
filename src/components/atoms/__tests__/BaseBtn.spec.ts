import { describe, it, expect, beforeEach } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import BaseBtn from "../BaseBtn.vue"

describe('BaseBtn', () => {
  let wrapper: ReturnType<typeof shallowMount>
  const createWrapper = () => shallowMount(BaseBtn, {
    global: {
      stubs: ['q-btn']
    }
  })

  beforeEach(() => {
    wrapper = createWrapper()
  })

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
