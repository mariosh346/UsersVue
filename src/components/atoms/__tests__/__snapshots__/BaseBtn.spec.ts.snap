// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseBtn > matches snapshot 1`] = `"<q-btn-stub color="primary" replace="false" disable="false" type="button" flat="false" outline="false" push="false" unelevated="false" square="false" rounded="false" glossy="false" fab="false" fabmini="false" nocaps="false" nowrap="false" dense="false" ripple="true" align="center" stack="false" stretch="false" loading="false" round="false" darkpercentage="false"></q-btn-stub>"`;
